packages/server/src/lib/access-helpers.js

```diff
@@ -25,36 +29,29 @@ async function requireAdminUser(req, res, next) {
     }
 
     // Depending on the request, an agency ID may be specified in zero or one of:
-    //  the query string: ?agency=...
+    //  a header: angecy-id
+    //  a query string: ?agency=...
     //  a route parameter :agency
     //  a route parameter :agencyId
     //  a body field named 'agency'
-    const queryAgency = Number(req.query.agency);
-    const paramAgency = Number(req.params.agency);
-    const paramAgencyId = Number(req.params.agencyId);
-    const bodyAgency = Number(req.body.agency);
+    const headerAgency = req.headers['agency-id'];
+    const queryAgency = req.query.agency;
+    const paramAgency = req.params.agency;
+    const paramAgencyId = req.params.agencyId;
+    // since the body is JSON, agency field will be a number, to keep everything the same type, convert it to a string
+    const bodyAgency = req.body.agency !== undefined && req.body.agency !== null ? req.body.agency.toString() : null;
 
-    let count = 0;
-    if (!Number.isNaN(queryAgency)) count += 1;
-    if (!Number.isNaN(paramAgency)) count += 1;
-    if (!Number.isNaN(paramAgencyId)) count += 1;
-    if (!Number.isNaN(bodyAgency)) count += 1;
+    const requestAgency = Number(bodyAgency || paramAgency || paramAgencyId || queryAgency || headerAgency);
 
-    if (count > 1) {
```

Comment:
Instead of only allowing one way to set the agency during a request, use precedence to decide which field to choose.

1. Body
2. Parmeter
3. Query
4. Header

Reply:
When I started the back end work, I also was going with a precedence scheme for a while. There are two reasons I moved away from it. 

1. If an API supports specifying some value as a URL parameter, I would expect I could modify a route to use that parameter, and have it work as expected. But it would not work as expected if I've forgotten/don't know that there is a body field taking precedence. If I specify a valid parameter, I wouldn't expect the routing to silently ignore it. Put a different way, it's a bad request that contradicts itself by asking for two different agencies in two different ways.
2. Probably more important, eliminating multiple ways of requesting simplified the testing. Allowing each request to specify an agency in up to 4 ways means a lot more test cases, and more complicated ones. As it is, many tests will need to be modified to use headers instead of query strings if that's the way we go.

Reply:
I agree. I added a header as another way of specifying the agency since it was easier for the client to send a header and also I removed the Bad Request restriction. If not the client will need to inspect every request to see how and where the agency should be passed.

Ill see if I can migrate this to a restul API if not Ill revert back the restriction to only allow one way of specifying the agency

Reply:
I think I could make it as easy to use query string as it is a header, and to avoid inspecting every request. We can go into details if we need to, but I think it's moot if we go more RESTful.